### DISCUSSION TITLE
Auto importing custom $fetch plugin globally

### DISCUSSION THREAD

**Discussion by omar-labana** [nuxt/nuxt#27063] [Questions] [answered]
Hey Nuxters,

I've been following the guide by @manniL on how to create a custom $fetch function by creating a plugin that provides my custom instance and I just need to expose it globally, any idea how to do it so it's just similar to $fetch?

```ts
export default defineNuxtPlugin({
  setup() {
    const { baseURL } = useRuntimeConfig().public;
    const store = useAuthenticationStore();

    const api = $fetch.create({
      baseURL,
      headers: {
        Locale: "en",
      },
      onRequest({ options }) {
        if (store.isAuthticated) {
          options.headers = options.headers || {};
          (
            options.headers as Record<string, string>
          ).Authorization = `Bearer ${store.session.token}`;
        }
      },
    });

    return {
      provide: {
        api,
      },
    };
  },
});
```

**Comment by TheAlexLichter** [comment, marked as answer]
From there you can go with `const { $api } = useNuxtApp()` and "just use it" but you need that extra line. 

Though you could also wrap that in a composable (if you want to use it as such) and auto-import that. Only works if you actually want to fetch in the nuxt context and *not* in an event handler (as explained [here](https://www.youtube.com/watch?v=njsGVmcWviY))

---

Also for the reference, [link to the guide](https://www.youtube.com/watch?v=jXH8Tr-exhI).
